File: LeetCode/26/main.cpp
class Solution {
public:
  int removeDuplicates(vector<int> &arr) {
    int n = arr.size();
    int left = 0;

    for (int i = 1; i < n; i++) {
      if (arr[i] != arr[left]) {
        left++;
        arr[left] = arr[i];
      }
    }

    return left + 1;
  }
};

File: LeetCode/26/main.py
class Solution:
    def removeDuplicates(self, arr: List[int]) -> int:
        # Overwrite using 2 pointers
        n = len(arr)
        ptr = 0
        for i in range(1, n, 1):
            # print('arr, ptr, i : ', arr, ptr, i)
            if arr[i] != arr[ptr]:
                ptr += 1
                # print('overwrite : ', i, ptr)
                arr[ptr] = arr[i]
        return ptr + 1
